src/lib.rs

```diff
@@ -770,6 +786,118 @@ impl Drop for Savefile {
     }
 }
 
+/// Abstraction for writing pcap savefiles in memory.  The `dump()` function can be used
+/// to retrieve a clone of the data currently in the savefile.
+#[cfg(not(any(windows, target_os="macos")))]
+pub struct Savememory {
+    handle: Unique<raw::pcap_dumper_t>,
+    shm_fd: libc::c_int,
+    shm_file: *mut libc::FILE,
+}
+
+#[cfg(not(any(windows, target_os="macos")))]
+impl Savememory {
+    /// Creates a new in-memory pcap file for the given pcap handle.
+    fn new(pcap_handle: *mut raw::pcap_t) -> Result<Savememory, Error> {
+        // Generate a random name so we don't collide
+        let mut name = vec![];
+        let mut name_rng = rand::thread_rng();
+        let mut name_gen = name_rng.gen_ascii_chars();
+        for _ in 0..20 {
```

Comment:
You should be able to do `for c in name_rng.gen_ascii_chars().take(20) { ... }`.